plugins/woocommerce/includes/admin/class-wc-admin-reports.php

```diff
@@ -211,22 +213,55 @@ public static function get_reports() {
 		 *         if ( is_admin() ) {
 		 *             // ...
 		 *
+		 * @since 2.1.0
+		 *
 		 * @param array $reports The associative array of reports.
 		 */
-		$reports = apply_filters( 'woocommerce_admin_reports', $reports );
-		// Backwards compatibility.
-		$reports = apply_filters_deprecated(
+		$filtered_reports = apply_filters( 'woocommerce_admin_reports', $reports );
+
+		/*
+		 * Check if there is any use of the legacy `woocommerce_admin_reports` filter to send a deprecation warning.
+		 * We will remove non-compliant entries twice.
+		 * First, here to check if there are any reports-specific, non-analytics changes to the original array.
+		 * Then send the non-sanitized array to the `woocommerce_reports_charts` filter to ensure 100% backward compatibility.
+		 * Then, we will sanitize again the result of both filters.
+		 */
+		$filtered_legacy_reports = $filtered_reports;
+		foreach ( $filtered_legacy_reports as $key => &$report_group ) {
+			// Remove entries not related to reports.
+			if ( ! isset( $report_group['reports'] ) ) {
+				unset( $filtered_legacy_reports[ $key ] );
+				continue;
+			}
+		}
+		// deep_compare_array_diff does not check additional entries.
+		$changed = ArrayUtil::deep_compare_array_diff( $reports, $filtered_legacy_reports, true ) ||
+					ArrayUtil::deep_compare_array_diff( $filtered_legacy_reports, $reports, true );
+		if ( $changed ) {
+			if ( WP_DEBUG && apply_filters( 'deprecated_hook_trigger_error', true ) ) { // phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingHookComment -- This is the use of the filter spefied in WP.
```

Comment:
I would recommend to add this conditional check right after the `apply_filters` line. Because if we aren't logging deprecation notices we don't need to use `$filtered_legacy_reports` or `$changed`, so we can save a bit of processing time.